### ISSUE TITLE
`transformItems` does not provide a reliable data structure for `items`

### ISSUE THREAD

**Issue by dhayab** [algolia/react-recommendations#62]
## Description

When using `useFrequentlyBoughtTogether` or `useRelatedProducts`, the `transformItems` property should allow me to retrieve the list of items, transform it and return a new list.

It appears the list of items (`items`) is not always an Array, making it impossible to reliably iterate over it or use any Array function (`map`, `filter`, etc.).

## Reproduction

In this sandbox, the Frequently Bought Together component is wrapped in a `connectStateResults` connector, allowing me to retrieve the ID of the first Algolia hit.

[**Preview →**](https://codesandbox.io/s/trusting-violet-fqb7q?file=/src/App.js)

**Steps**

Just open the console and see subsequents calls to `transformItems` providing a wrong structure as `items`.

## Expected behavior

`items` is always an Array.

## Environment

- OS: macOS
- Browser: Chrome
- Library version:
  - @algolia/recommend: 4.12.0
  - @algolia/recommend-react: 1.1.0
  - react-instantsearch-dom: 6.0.0

**Comment by Haroenv** [comment]
relevant code: https://github.com/algolia/recommend/blob/e0318420a12100658e9d8a04752f6464d874e227/packages/recommend-core/src/getFrequentlyBoughtTogether.ts#L39

is it hits that can be null or something else?

**Comment by dhayab** [comment]
From what I see in the console, hits can be a reference to `transformItems()`.

**Comment by evanwinter** [comment]
Any progress on this? I'm running into this too.

```
const { recommendations } = useRelatedProducts({
    recommendClient,
    indexName,
    objectIDs,
    queryParameters: { filters },
    transformItems: (items) => {
        console.log(items);
        return items;
    },
});
```

<img width="464" alt="Screen Shot 2022-10-13 at 11 32 45 AM" src="https://user-images.githubusercontent.com/19195468/195654237-b627a163-8a68-4a7d-b096-b3323e131569.png">


On the first cycle, `items` is an Array but every cycle after that it looks like it's a reference to `transformItems`

**Comment by francoischalifour** [comment]
This happens because we try to stabilize the `transformItems` prop: https://github.com/algolia/recommend/blob/d8e63758684241ea24d2abb5465e47b38cb32225/packages/recommend-react/src/useTrendingItems.ts#L27

But this function stores a reference to the function because of the lazy initialization syntax: https://github.com/algolia/recommend/blob/d8e63758684241ea24d2abb5465e47b38cb32225/packages/recommend-react/src/useStableValue.ts#L5

We should be safe using a regular `useState(value)` here.

However to correctly stabilize the function we should probably store it in a ref.

// @algolia/recommend-frontend

**Comment by marialungu** [comment]
closed in https://github.com/algolia/recommend/pull/110, available with recommend [v1.8.0](https://github.com/algolia/recommend/releases/tag/v1.8.0)
